src/scaling/scaling.jl

```diff
@@ -2,7 +2,7 @@ export ScaledInterpolation, eachvalue
 
 import Base: iterate
 
-struct ScaledInterpolation{T,N,ITPT,IT,RT} <: AbstractInterpolationWrapper{T,N,ITPT,IT}
+struct ScaledInterpolation{T,N,ITPT,IT,RT<:NTuple{N,AbstractRange}} <: AbstractInterpolationWrapper{T,N,ITPT,IT}
```

Comment:
Is this necessary?

Reply:
The current added code assumes that ranges here is GPU compatible.
I think we’d better add this restriction to constructor.

Reply:
The only problem is that this is a potentially breaking change, although I'm not sure if I have a concrete example yet.

A non-breaking change would be to create this type constant and use that.
```julia
const RangedScaledInterpolation{T, N, ITPT, IT, RT <: NTuple{N,AbstractRange}} = ScaledInterpolation{T,N,ITPT,IT,RT}
```

Reply:
In fact, all methods belows assumes that `ranges` all have `step` defined, so I think this change should be safe.
But of course we can discuss this in another thread. I'll move this restriction to `adapt` side.